### ISSUE TITLE
scale_x_date limits not subsetting data

### ISSUE THREAD

**Issue by liouville1** [hadley/ggplot2#1090]
Posted on Stack Overflow:

http://stackoverflow.com/questions/29900106/ggplot2-bug-ggplot-showing-not-dropping-data-outside-limits-of-scale#29900106

I've noticed that scale_x_date is displaying data outside of the limits specified in the scale. See example below. Why am I seeing points before 2009/01/01 and after 2015/01/01 in the graph if the limits are set to these values?
	
	library(ggplot2)
	library(scales)
    set.seed(100)
	z <- seq.Date(as.Date("2008/12/1"), as.Date("2015/12/14"), "day")
	l <- expand.grid(z, c("a", "b", "c"))
	w <- data.frame(x= l[, 1], t = l[, 2])
	w$val <- runif(nrow(w))
	ggplot(data=w, aes_string(x="x", y="val"))+scale_x_date(
	  labels = date_format("%m/%d/%Y"),
	  limits= c(as.Date("2009/1/1"), as.Date("2015/1/1")),
	  breaks = "1 year")+
	  geom_point(aes(color = t))

![resulting plot][1]

  [1]: http://i.stack.imgur.com/9CcnZ.png

    n <- as.Date("2009/1/1")
    ggplot(w,aes(x,val))+geom_point()+
    scale_x_date(limits=c(n,NA))+
      geom_vline(xintercept=as.numeric(n),colour="red")

![resulting plot][2]

  [2]: http://i.stack.imgur.com/1E6Dv.png

**Comment by hadley** [comment]
Limits are automatically expanded using the expand parameter.

**Comment by liouville1** [comment]
Just to be clear, the limits of the scales are controlled by the expand parameter, but the limits of the data should not be affected. At least this is the behavior when using scale_x_continuous. Currently scale_x_date is functioning differently than scale_x_continuous.

See below

With scale_x_continuous

           gplot(w,aes(as.numeric(x),val))+geom_point()+
           scale_x_continuous(limits=c(as.numeric(n),NA))+
           geom_vline(xintercept=as.numeric(n),colour="red")

![resulting plot][1]
[1]: http://i.stack.imgur.com/1heOc.png

with scale_x_date

       ggplot(w,aes(x,val))+geom_point()+
       scale_x_date(limits=c(n,NA))+
        geom_vline(xintercept=as.numeric(n),colour="red")

![resulting plot][2]

[2]: http://i.stack.imgur.com/1E6Dv.png

**Comment by hadley** [comment]
Oh, hmmm, that's a more compelling bug.

**Comment by hadley** [comment]
Here's a somewhat simpler reproducible example:

```R
library(ggplot2)

df <- data.frame(
  x = as.Date("2000-01-01") + 0:50,
  y = 1, 
  z = 0:50
)
ggplot(df, aes(x, y, colour = z >= 1)) +
  geom_point()

ggplot(df, aes(x, y, colour = z >= 1)) +
  geom_point() + 
  scale_x_date(limits = c(df$x[[2]], NA))

ggplot(df, aes(as.numeric(x), y, colour = z >= 1)) +
  geom_point() + 
  scale_x_continuous(limits = c(df$x[[2]], NA))
```

**Comment by hadley** [comment]
Fixed in ae1303f8d46bdf5e8e96e32df2dfec53f76bdab0
